scripts/build-image.sh

```diff
@@ -25,21 +25,40 @@ platform="${3}"
 output="${4}"
 builder="${5}"
 
+shift 5
+
+registries=("${@}")
+
 image_tag="$("${root_dir}/scripts/make-image-tag.sh" "${image_dir}")"
 
-check_tag_exists() {
+tag_args=()
+for registry in "${registries[@]}" ; do
+  tag_args+=(--tag "${registry}/${image_name}:${image_tag}")
+done
+
+check_image_tag() {
   if [ -n "${MAKER_CONTAINER+x}" ] ; then
-    crane ls "${image_name}" --verbose | grep -q "${image_tag}"
+    crane digest "${1}" || (echo "error: crane returned $?" ; return 1)
```

Comment:
@borkmann I think this should fix the issue with unexpected rebuilds, it's a more monolithic check.

What I was seeing from the logs is `crane ls` returning all tags (according to verbose logs), but `grep` failing to pick them up. My general theory is that `complex-remote-call | grep` is never a good idea, another fix would be to redirect to a file or something like that, but there is simple option with `crane digest` here.